src/MessageBus.php

```diff
@@ -61,75 +61,49 @@ public function deactivate(ActionEventListenerAggregate $plugin): void
         $plugin->detach($this->getActionEventEmitter());
     }
 
-    /**
-     * @param mixed $message
-     * @param ActionEvent $actionEvent
-     */
-    protected function initialize($message, ActionEvent $actionEvent): void
-    {
-        $actionEvent->setParam(self::EVENT_PARAM_MESSAGE, $message);
-        $actionEvent->setParam(self::EVENT_PARAM_MESSAGE_HANDLED, false);
-
-        if ($message instanceof HasMessageName) {
-            $actionEvent->setParam(self::EVENT_PARAM_MESSAGE_NAME, $message->messageName());
-        }
-
-        $actionEvent->setName(self::EVENT_INITIALIZE);
-
-        $this->trigger($actionEvent);
-
-        if ($actionEvent->getParam(self::EVENT_PARAM_MESSAGE_NAME) === null) {
-            $actionEvent->setName(self::EVENT_DETECT_MESSAGE_NAME);
-
-            $this->trigger($actionEvent);
-
-            if ($actionEvent->getParam(self::EVENT_PARAM_MESSAGE_NAME) === null) {
-                $actionEvent->setParam(self::EVENT_PARAM_MESSAGE_NAME, $this->getMessageName($message));
-            }
-        }
-    }
-
-    protected function handleException(ActionEvent $actionEvent, \Throwable $ex): void
-    {
-        $failedPhase = $actionEvent->getName();
-
-        $actionEvent->setParam(self::EVENT_PARAM_EXCEPTION, $ex);
-        $this->triggerError($actionEvent);
-        $this->triggerFinalize($actionEvent);
-
-        //Check if a listener has removed the exception to indicate that it was able to handle it
-        if ($ex = $actionEvent->getParam(self::EVENT_PARAM_EXCEPTION)) {
-            $actionEvent->setName($failedPhase);
-            throw MessageDispatchException::failed($actionEvent, $ex);
-        }
-    }
-
-    protected function trigger(ActionEvent $actionEvent): void
-    {
-        $this->getActionEventEmitter()->dispatch($actionEvent);
-
-        if ($actionEvent->propagationIsStopped()) {
-            throw new RuntimeException('Dispatch has stopped unexpectedly.');
-        }
-    }
-
-    protected function triggerError(ActionEvent $actionEvent): void
-    {
-        $actionEvent->setName(self::EVENT_HANDLE_ERROR);
-
-        $this->getActionEventEmitter()->dispatch($actionEvent);
```

Comment:
From the outside there doesn't change anything. It's still like

```
try {
    $commandBus->dispatch($command);
} catch (DispatchException $e) {
    echo $e->getPrevious()->getMessage();
}
```

If you want to have a listener on it, things changed a little:

```
$commandBus->getActionEventEmitter()->attachListener(
    CommandBus::EVENT_FINALIZE,
    function (ActionEvent $actionEvent) {
        if ($ex = $actionEvent->getParam(CommandBus::EVENT_PARAM_EXCEPTION) {
            // do something
        }
    }
```